packages/gatsby/src/utils/page-data.ts

```diff
@@ -0,0 +1,69 @@
+import fs from "fs-extra"
+import path from "path"
+import { IGatsbyPage } from "../redux/types"
+import { store } from "../redux"
+
+interface IPageDataContext {
+  publicDir: string
+}
+
+type IPagePath = IGatsbyPage["path"]
+
+export const fixedPagePath = (pagePath: IPagePath): IPagePath =>
+  pagePath === `/` ? `index` : pagePath
+
+const getFilePath = (
+  { publicDir }: IPageDataContext,
+  pagePath: IPagePath
+): IPagePath =>
+  path.join(publicDir, `page-data`, fixedPagePath(pagePath), `page-data.json`)
+
+export const read = async (
+  { publicDir }: IPageDataContext,
+  pagePath: IPagePath
+): Promise<IGatsbyPage> => {
```

Comment:
I tried to investigate quickly in the codebase what seems to be the type of information returned by this function but I'm not entirely sure here.

Should we have something super generic at the moment like `any` or is it a safe assumption that the asynchronous function returns a promise of `IGatsbyPage`?